[1.3.0] sbt cannot compile compiler bridge if you point at zinc's compiler bridge project

## steps

1. Get a Docker image, fresh machine, or remove existing compiler bridge.

```
rm -rf ~/.sbt/1.0/zinc/org.scala-sbt/**-1.3.0**
rm -rf ~/.sbt/boot/scala-2.12.8/org.scala-sbt/sbt/1.3.0-LOCAL-20190512/
```

2. Make a fresh sbt. Let's say 1.3.0-LOCAL-20190512.
3. open sbt/zinc project and `compilerBridge211/compile`

## problem

```
sbt:zinc Root> compilerBridge211/compile
[info] Compiling 24 Scala sources to /Users/eed3si9n/work/sbt-modules/zinc/internal/compiler-bridge/target-2.11/scala-2.11/classes ...
[info] Non-compiled module 'compiler-bridge_2.11' for Scala 2.11.12. Compiling...
/var/folders/hg/2602nfrs2958vnshglyl3srw0000gn/T/sbt_ad78e201/xsbt/API.scala:12: error: object api is not a member of package xsbti
import xsbti.api._
             ^
/var/folders/hg/2602nfrs2958vnshglyl3srw0000gn/T/sbt_ad78e201/xsbt/CallbackGlobal.scala:11: error: object compile is not a member of package xsbti
import xsbti.compile._
             ^
/var/folders/hg/2602nfrs2958vnshglyl3srw0000gn/T/sbt_ad78e201/xsbt/CallbackGlobal.scala:10: error: object AnalysisCallback is not a member of package xsbti
import xsbti.{ AnalysisCallback, Severity }
       ^
```

## expectation

it works.

## note

This situation happens when you use sbt 1.3.0-M3 for sbt/zinc since Scala 2.11 compiler bridge will be the first 2.11 it will compile. A weird workaround might be to first compile helloworld using 2.11.

--- a/main/src/main/scala/sbt/internal/LMCoursier.scala
+++ b/main/src/main/scala/sbt/internal/LMCoursier.scala
@@ -9,7 +9,12 @@ package sbt
 package internal
 
 import java.io.File
-import lmcoursier.definitions.{ Classifier, Configuration => CConfiguration, CacheLogger }
+import lmcoursier.definitions.{
+  Classifier,
+  Configuration => CConfiguration,
+  CacheLogger,
+  Project => CProject
+}
 import lmcoursier._
 import sbt.librarymanagement._
 import Keys._
@@ -45,7 +50,12 @@ private[sbt] object LMCoursier {
         val rs = resolversTask.value
         val scalaOrg = scalaOrganization.value
         val scalaVer = scalaVersion.value
-        val interProjectDependencies = csrInterProjectDependencies.value
+        val interProjectDependencies0 = csrInterProjectDependencies.value.toVector
+        // during the resolution of sbt artifacts, such as compiler bridge source
+        // interproject dependencies should not be used (otherwise you can't compile sbt/zinc using sbt)
+        val interProjectDependencies: Vector[CProject] =
+          if (sbtClassifiers) Vector()
+          else interProjectDependencies0
         val excludeDeps = Inputs.exclusions(
           allExcludeDependencies.value,
           scalaVer,
@@ -71,7 +81,7 @@ private[sbt] object LMCoursier {
         Classpaths.warnResolversConflict(rs, log)
         CoursierConfiguration()
           .withResolvers(rs.toVector)
-          .withInterProjectDependencies(interProjectDependencies.toVector)
+          .withInterProjectDependencies(interProjectDependencies)
           .withFallbackDependencies(fallbackDeps.toVector)
           .withExcludeDependencies(
             excludeDeps.toVector.map {

